src/NuGet.Clients/NuGet.CommandLine/Commands/ProjectFactory.cs

```diff
@@ -1066,7 +1066,20 @@ private static string RemoveExtension(string path)
         private IEnumerable<Packaging.IPackageFile> GetTransformFiles(PackageReaderBase package)
         {
             var groups = package.GetContentItems();
-            return groups.SelectMany(g => g.Items).Where(IsTransformFile).Select(f => new Packaging.PhysicalPackageFile() { TargetPath = f });
+            return groups.SelectMany(g => g.Items).Where(IsTransformFile).Select(f =>
+            {
+                Packaging.PhysicalPackageFile file = new Packaging.PhysicalPackageFile()
+                {
+                    TargetPath = f,
+                    MemoryStream = new MemoryStream()
+                };
+
+                XElement element = XElement.Load(package.GetStream(f));
+                element.Save(file.MemoryStream);
+                file.MemoryStream.Seek(0, SeekOrigin.Begin);
```

Comment:
I think it'd better if `file` would get created after the memory stream is fully initialized.